### PR TITLE
[KYUUBI #5331] Spark engine should throw an exception when it fails to start

### LINKED ISSUE #5331: [TASK][EASY][Bug] Spark Engine should throw an exception when the engine fails to start.
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

In the main method of SparkSQLEngine, exceptions are caught but not thrown. If the engine fails to start, the k8s status is displayed as finished, and the kyuubi server will not exit until the engine initialization times out.
d45cf19d59ae6934430910300a28fff

### Affects Version(s)

master/1.7.1

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by ASiegeLion**
### _Why are the changes needed?_

Close #5331

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No.

### CODE REVIEW

**Comment by pan3793** [comment]
@ASiegeLion I'm going to prepare the 1.8.0 RC0 late this week, do you have time to finish this PR so that we can ship it 1.8.0? (PS: I don't think it's a blocking issue since it's not a regression and is not common enough)

**Comment by ASiegeLion** [comment]
> @ASiegeLion I'm going to prepare the 1.8.0 RC0 late this week, do you have time to finish this PR so that we can ship it in 1.8.0? (PS: I don't think it's a blocking issue since it's not a regression and is not common enough)

Sorry, I didn't notice that CI failed. How can I find the reason for the failure?
image

**Comment by pan3793** [comment]
Are you possible to run UT locally?

**Comment by ASiegeLion** [comment]
> Are you possible to run UT locally?
ok, I run the command  : ./build/mvn clean test -pl :kyuubi-spark-sql-engine_2.12, 
image
image

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala [hunk 1]
```diff
@@ -379,7 +381,11 @@ object SparkSQLEngine extends Logging {
               s" The `${ENGINE_INIT_TIMEOUT.key}` is ($initTimeout ms) " +
               s" and submitted at $submitTime.",
             i)
-        case t: Throwable => error(s"Failed to instantiate SparkSession: ${t.getMessage}", t)
+          throw i
```
Comment (wForget) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:384]: Wrap exception with errorMsg.

**Review by wForget — APPROVED** [review]

**Comment by pan3793** [comment]
thanks all, merged to master/1.8

**Pull request closed without merging**